app/portainer/views/devices/import/importDeviceController.js

```diff
@@ -1,140 +1,141 @@
 import { PortainerEndpointCreationTypes } from 'Portainer/models/endpoint/models';
-import { configureDevice, getProfiles } from "Portainer/hostmanagement/fdo/fdo.service";
+import { configureDevice, getProfiles } from 'Portainer/hostmanagement/fdo/fdo.service';
 
 angular
   .module('portainer.app')
-  .controller('ImportDeviceController', function ImportDeviceController(
-    $async,
-    $q,
-    $scope,
-    $state,
-    EndpointService,
-    GroupService,
-    TagService,
-    Notifications,
-    Authentication,
-    FileUploadService,
-  ) {
-    $scope.state = {
-      actionInProgress: false,
-      voucherUploading: false,
-      voucherUploaded: false,
-      deviceID: '',
-      allowCreateTag: Authentication.isAdmin(),
-    };
-
-    $scope.formValues = {
-      DeviceName: '',
-      DeviceProfile: '',
-      GroupId: 1,
-      TagIds: [],
-      VoucherFile: null,
-      PortainerURL: '',
-    };
-
-    $scope.profiles = [];
-
-    $scope.onVoucherFileChange = function (file) {
-      if (!file) {
-        return;
-      }
-
-      $scope.state.voucherUploading = true;
-
-      FileUploadService.uploadOwnershipVoucher(file)
-        .then(function success(response) {
-          $scope.state.voucherUploading = false;
-          $scope.state.voucherUploaded = true;
-          $scope.deviceID = response.data.guid;
-        })
-        .catch(function error(err) {
-          $scope.state.voucherUploading = false;
-          Notifications.error('Failure', err, 'Unable to upload Ownership Voucher');
-        });
-    };
-
-    $scope.onCreateTag = function onCreateTag(tagName) {
-      return $async(onCreateTagAsync, tagName);
-    };
-
-    async function onCreateTagAsync(tagName) {
-      try {
-        const tag = await TagService.createTag(tagName);
-        $scope.availableTags = $scope.availableTags.concat(tag);
-        $scope.formValues.TagIds = $scope.formValues.TagIds.concat(tag.Id);
-      } catch (err) {
-        Notifications.error('Failure', err, 'Unable to create tag');
-      }
-    }
-
-    $scope.createEndpointAndConfigureDevice = function () {
-      return $async(async () => {
-        $scope.state.actionInProgress = true;
-
+  .controller(
+    'ImportDeviceController',
+    function ImportDeviceController($async, $q, $scope, $state, EndpointService, GroupService, TagService, Notifications, Authentication, FileUploadService) {
+      $scope.state = {
+        actionInProgress: false,
+        vouchersUploading: false,
+        vouchersUploaded: false,
+        deviceIDs: [],
+        allowCreateTag: Authentication.isAdmin(),
+      };
+
+      $scope.formValues = {
+        DeviceName: '',
+        DeviceProfile: '',
+        GroupId: 1,
+        TagIds: [],
+        VoucherFiles: null,
+        PortainerURL: '',
+        Suffix: 1,
+      };
+
+      $scope.profiles = [];
+
+      $scope.onVoucherFilesChange = function () {
+        $scope.state.vouchersUploading = true;
```

Comment:
we have to validate that there is a valid files array
```suggestion
        if ($scope.formValues.VoucherFiles.length < 1) {
             return;
        }
        $scope.state.vouchersUploading = true;
```